drivers/adc/adc_demo/adc_demo.c

```diff
@@ -85,13 +112,141 @@ int32_t adc_demo_remove(struct adc_demo_desc *desc)
 	return SUCCESS;
 }
 
-/***************************************************************************//**
+/**********************************************************************//**
+ * @brief close all channels
+ * @param dev - physical instance of an adc device
+ * @return SUCCESS in case of success.
+**************************************************************************/
+int32_t close_adc_channels(void* dev)
+{
+	if(!dev)
+		return -ENODEV;
+
+	struct adc_demo_desc *desc = dev;
```

Comment:
leave the variable definition at function start just don't do the attribution operation until the check for !dev